ui/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter.js

```diff
@@ -0,0 +1,370 @@
+        : null}
+  `
+
+const Filter = withListenTo(
+  class Filter extends React.Component {
+    filterAttribute
+    filterInput
+    constructor(props) {
+      super(props)
+      const comparator = props.model.get('comparator') || 'CONTAINS'
+      let attribute = props.model.get('type') || 'anyText'
+      if (
+        props.includedAttributes &&
+        !props.includedAttributes.includes(attribute)
+      ) {
+        attribute = props.includedAttributes[0]
+      }
+
+      this.state = {
+        comparator,
+        attribute,
+        editing: props.editing,
+        type: metacardDefinitions.metacardTypes[attribute].type,
+        filteredAttrs: this.getFilteredAttributeList(),
+      }
+      props.model.set('type', attribute)
+      props.model.set('comparator', comparator)
+    }
+
+    componentDidMount = () => {
+      this.props.listenTo(
+        this.props.model,
+        'change:comparator',
+        this.updateComparator
+      )
+    }
+
+    componentWillReceiveProps = ({ editing }) => {
+      this.setState({ editing })
+    }
+
+    render() {
+      const filterDropdownModel = new DropdownModel({
+        value: this.state.comparator,
+      })
+      const filterComparator = new FilterComparatorDropdownView({
+        model: filterDropdownModel,
+        modelForComponent: this.props.model,
+      })
+      this.filterAttribute = DropdownView.createSimpleDropdown({
+        list: this.state.filteredAttrs,
+        defaultSelection: [this.state.attribute],
+        hasFiltering: true,
+      })
+
+      this.props.listenTo(
+        this.filterAttribute.model,
+        'change:value',
+        this.updateAttribute
+      )
+
+      this.determineInput()
+      if (this.state.editing) {
+        this.filterAttribute.turnOnEditing()
+        filterComparator.turnOnEditing()
+        const property =
+          this.filterInput.model instanceof ValueModel
+            ? this.filterInput.model.get('property')
+            : this.filterInput.model
+        property.set('isEditing', true)
+      } else {
+        this.filterAttribute.turnOffEditing()
+        filterComparator.turnOffEditing()
+        const property =
+          this.filterInput.model instanceof ValueModel
+            ? this.filterInput.model.get('property')
+            : this.filterInput.model
+        property.set(
+          'isEditing',
+          this.props.isForm === true || this.props.isFormBuilder === true
+        )
+      }
+      return (
+        <React.Fragment>
+          <div className="filter-rearrange">
+            <span className="cf cf-sort-grabber" />
+          </div>
+          <button className="filter-remove is-negative">
+            <span className="fa fa-minus" />
+          </button>
+          <div
+            className="filter-attribute"
+            data-help="Property to compare against."
+          >
+            <MarionetteRegionContainer view={this.filterAttribute} />
```

Comment:
Could you have filterAttribute be its own component and pass down necessary props?

Reply:
Could you be more specific? filterAttribute is already its own marionette component.

Reply:
I was thinking you could move both filterInput and filterAttribute into their own React components (MarionetteRegionContainers wrapping filterInput/filterAttribute), then pass down state/props from this component to the things you need to the component like these model changes https://github.com/codice/ddf/pull/5099/files#diff-3d119cbaa55f40321e65dbdb07e8f774R118